plugins/modules/firewall.py

```diff
@@ -262,6 +262,12 @@ def _amend_rules(remote_rules: list, local_rules: list) -> list:
             for field in linode_firewall_rule_spec:
                 if field not in local_rule and field in remote_rule:
                     local_rule[field]=remote_rule[field]
+            for ip in ['ipv6','ipv4']:
+                if ip not in local_rule.get('addresses',{}):
+                    remote_addresses=remote_rule.get('addresses',{})
+                    local_addresses=local_rule.get('addresses',{})
+                    local_addresses[ip]=remote_addresses.get(ip,[])
+                    local_rule['addresses']=local_addresses
```

Comment:
Small change to improve readability a bit
```suggestion
            for ip in ['ipv6', 'ipv4']:
                if ip in local_rule.get('addresses', {}):
                    continue
                    
                remote_addresses=remote_rule.get('addresses',{})
                local_addresses=local_rule.get('addresses',{})
                local_addresses[ip]=remote_addresses.get(ip,[])
                local_rule['addresses']=local_addresses
```